base/shell/explorer/traywnd.cpp

```diff
@@ -264,6 +264,115 @@ class CStartButton
 
 };
 
+static const WCHAR szTrayShowDesktopButton[] = L"TrayShowDesktopButtonWClass";
+
+// The 'Show Desktop' button at edge of taskbar
+class CTrayShowDesktopButton :
+    public CWindowImpl<CTrayShowDesktopButton, CWindow, CControlWinTraits>
+{
+    LONG m_nClickedTime;
+
+public:
+    DECLARE_WND_CLASS_EX(szTrayShowDesktopButton, CS_HREDRAW | CS_VREDRAW, COLOR_3DFACE)
+
+    CTrayShowDesktopButton() : m_nClickedTime(0)
+    {
+    }
+
+    INT WidthOrHeight() const
+    {
+        INT cxy = GetSystemMetrics(SM_CXEDGE);
+        return (cxy ? (2 * cxy) : 6);
+    }
+
+    HRESULT Create(HWND hwndParent)
+    {
+        DWORD style = WS_CHILD | WS_VISIBLE | WS_CLIPSIBLINGS;
+        ATL::CWindowImpl<CTrayShowDesktopButton, CWindow, CControlWinTraits>::
+            Create(hwndParent, NULL, NULL, style);
+        if (!m_hWnd)
+            return E_FAIL;
+
+        if (::SetWindowTheme(m_hWnd, L"Start", NULL) != S_OK)
+            ::SetWindowTheme(m_hWnd, L"Explorer", NULL);
+        return S_OK;
+    }
+
+    LRESULT OnClick(UINT uMsg, WPARAM wParam, LPARAM lParam, BOOL& bHandled)
+    {
+        LONG nTime0 = m_nClickedTime;
+        LONG nTime1 = ::GetMessageTime();
+        if (nTime1 - nTime0 >= 800) // 0.8sec
```

Comment:
Don't hesitate to write comments in code, they exist for a reason. In particular, what is the reason for this thing.

Reply:
Added comments.